x-pack/plugin/core/src/test/java/org/elasticsearch/index/engine/FrozenIndexTests.java

```diff
@@ -241,6 +273,24 @@ public void testFreezePattern() throws ExecutionException, InterruptedException
     }
 
     public void testCanMatch() throws ExecutionException, InterruptedException, IOException {
+        if (randomBoolean()) {
+            readerWrapper.set(reader -> new FilterDirectoryReader(reader, new FilterDirectoryReader.SubReaderWrapper() {
```

Comment:
AFAICS, this does not provoke a failure without the fix, can we somehow make it call `getCoreCacheHelper`? Or maybe just make the wrapper throw to proof that it is not used for the can match phase?

Reply:
Ok, I pushed https://github.com/elastic/elasticsearch/pull/64742/commits/9581a0cd43f5c7f623c3673dd9c10a749c5911cb